lib/ain-evm/src/block.rs

```diff
@@ -182,136 +191,146 @@ impl BlockService {
-
-                let gas_ratio = if block.header.gas_limit == U256::zero() {
-                    f64::default() // empty block
-                } else {
-                    u64::try_from(block.header.gas_used)? as f64
-                        / u64::try_from(block.header.gas_limit)? as f64 // safe due to check
-                };
-
-                base_fee_per_gas.push(base_fee);
-                gas_used_ratio.push(gas_ratio);
-
-                Ok::<_, EVMError>((base_fee_per_gas, gas_used_ratio))
-            },
-        )?;
-
-        let reward = if priority_fee_percentile.is_empty() {
-            None
-        } else {
-            let mut eip_transactions = Vec::new();
-
-            for block in blocks {
-                let mut block_eip_transaction = Vec::new();
-                for tx in block.transactions {
-                    match tx {
-                        TransactionAny::Legacy(_) | TransactionAny::EIP2930(_) => {
-                            continue;
-                        }
-                        TransactionAny::EIP1559(t) => {
-                            block_eip_transaction.push(t);
-                        }
-                    }
-                }
-                block_eip_transaction
-                    .sort_by(|a, b| a.max_priority_fee_per_gas.cmp(&b.max_priority_fee_per_gas));
-                eip_transactions.push(block_eip_transaction);
+        // Validate reward_percentile input
+        if reward_percentile.len() > MAX_REWARD_PERCENTAGE {
+            return Err(format_err!(
+                "List of percentile values exceeds maximum allowed size of {}",
+                MAX_REWARD_PERCENTAGE,
+            )
+            .into());
+        }
+        let mut prev_percentile = 0;
+        for percentile in &reward_percentile {
+            if *percentile > MAX_REWARD_PERCENTAGE {
+                return Err(format_err!(
+                    "Percentile value more than inclusive range of {}",
+                    MAX_REWARD_PERCENTAGE,
+                )
+                .into());
             }
+            if prev_percentile > *percentile {
+                return Err(format_err!(
+                    "List of percentile values are not monotonically increasing"
+                )
+                .into());
+            }
+            prev_percentile = *percentile;
+        }
 
-            /*
-                TODO: assumption here is that max priority fee = priority fee paid, however
-                priority fee can be lower if gas costs hit max_fee_per_gas.
-                we will need to check the base fee paid to get the actual priority fee paid
-            */
-
-            let mut reward = Vec::new();
-
-            for block_eip_tx in eip_transactions {
-                if block_eip_tx.is_empty() {
-                    reward.push(vec![U256::zero()]);
-                    continue;
-                }
+        let mut blocks = Vec::new();
```

Comment:
Restoring previous init
```suggestion
        let mut blocks = Vec::with_capacity(block_count);
```

Reply:
Resolved in commit d14f60b0db